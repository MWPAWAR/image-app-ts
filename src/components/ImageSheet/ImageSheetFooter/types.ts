import { ImageTypes } from '../../types'

export interface ImageSheetFooterProps {
  selectedImage: ImageTypes
  toggleDrawer: () => void
}
