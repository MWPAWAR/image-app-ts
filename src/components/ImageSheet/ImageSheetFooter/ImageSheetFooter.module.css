.imageSheetFooter {
  align-items: center;
  display: flex;
  flex-direction: column;
  margin-top: 30px;
}

.button {
  margin: 2%;
  width: 80%;
}

.deleteBtn {
  background-color: #fff;
  border: 1px solid #ff4d4f;
  border-radius: 5px;
  color: #ff4d4f;
  cursor: pointer;
  padding: 10px;
  width: 100%;
}

.backBtn {
  background-color: #fff;
  border: 1px solid #d9d9d9;
  border-radius: 5px;
  cursor: pointer;
  padding: 10px;
  width: 100%;
}
