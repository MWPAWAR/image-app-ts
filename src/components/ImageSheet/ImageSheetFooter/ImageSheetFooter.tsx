import { useAppDispatch } from '../../customHooks/hooks'
import imageSheetFooterStyles from './ImageSheetFooter.module.css'
import { deleteImg } from '../../../store/images/images'
import { resetSelectedImg } from '../../../store/imageUiState/imageUiState'
import { ImageSheetFooterProps } from './types'
import { getString } from '../../../i18n'

const ImageDeleteBtn = ({
  selectedImage,
  toggleDrawer,
}: ImageSheetFooterProps) => {
  const dispatch = useAppDispatch()
  const handleBackBtnClick = (): void => {
    dispatch(resetSelectedImg())
    toggleDrawer()
  }
  const handleDeleteClick = (): void => {
    dispatch(deleteImg({ id: selectedImage.id }))
    dispatch(resetSelectedImg())
    toggleDrawer()
  }

  return (
    <div className={imageSheetFooterStyles.imageSheetFooter}>
      <div className={imageSheetFooterStyles.button}>
        <button
          className={imageSheetFooterStyles.deleteBtn}
          onClick={handleDeleteClick}
        >
          {getString('imageSheet.deleteBtnLabel')}
        </button>
      </div>
      <div className={imageSheetFooterStyles.button}>
        <button
          className={imageSheetFooterStyles.backBtn}
          onClick={handleBackBtnClick}
        >
          {getString('imageSheet.backBtnLabel')}
        </button>
      </div>
    </div>
  )
}

export default ImageDeleteBtn
