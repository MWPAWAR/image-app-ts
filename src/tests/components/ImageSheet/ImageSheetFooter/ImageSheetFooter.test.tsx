import { MemoryRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import { render } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import ImageSheetFooter from '../../../../components/ImageSheet/ImageSheetFooter/ImageSheetFooter'
import { imageMock1 } from '../../../mocks/image'
import { toggleDrawer } from '../../../test-utils'

describe('ImageSheetFooter', () => {
  const mockStore = configureStore([])
  const store = mockStore({})
  jest.spyOn(store, 'dispatch')

  const ImageSheetFooterComponent = () => (
    <MemoryRouter>
      <Provider store={store}>
        <ImageSheetFooter
          selectedImage={imageMock1}
          toggleDrawer={toggleDrawer}
        />
      </Provider>
    </MemoryRouter>
  )

  it('Matches ImageDeleteBtn snapshot', () => {
    const tree = renderer.create(<ImageSheetFooterComponent />)
    expect(tree).toMatchSnapshot()
  })

  it('Calls dispatch twice if user clicks on delete button', async () => {
    const { getByText } = render(<ImageSheetFooterComponent />)
    await userEvent.click(getByText('Delete'))
    expect(store.dispatch).toBeCalledTimes(2)
    expect(store.dispatch).toBeCalledWith({
      type: 'imageUiState/resetSelectedImg',
    })
    expect(toggleDrawer).toHaveBeenCalled()
    expect(store.dispatch).toBeCalledWith({
      type: 'images/deleteImg',
      payload: { id: imageMock1.id },
    })
  })

  it('Calls dispatch if user clicks on back button', async () => {
    const { getByText } = render(<ImageSheetFooterComponent />)
    await userEvent.click(getByText('Back'))
    expect(store.dispatch).toBeCalledTimes(1)
    expect(store.dispatch).toBeCalledWith({
      type: 'imageUiState/resetSelectedImg',
    })
    expect(toggleDrawer).toHaveBeenCalled()
  })
})
