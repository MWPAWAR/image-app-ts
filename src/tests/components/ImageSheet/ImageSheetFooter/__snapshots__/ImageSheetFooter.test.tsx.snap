// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageSheetFooter Matches ImageDeleteBtn snapshot 1`] = `
<div
  className="imageSheetFooter"
>
  <div
    className="deleteBtn"
  >
    <button
      className="button"
      onClick={[Function]}
    >
      Delete
    </button>
  </div>
  <div
    className="backBtn"
  >
    <button
      className="button"
      onClick={[Function]}
    >
      Back
    </button>
  </div>
</div>
`;
